lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -99,7 +100,34 @@ def scrub(transcript)
           gsub(/("number\\":\\")\d+/, '\1[FILTERED]').
           gsub(/("cvv\\":\\")\d+/, '\1[FILTERED]').
           gsub(/("cryptogram\\":\\")\w+/, '\1[FILTERED]').
-          gsub(/(source\\":\{.*\\"token\\":\\")\d+/, '\1[FILTERED]')
+          gsub(/(source\\":\{.*\\"token\\":\\")\d+/, '\1[FILTERED]').
+          gsub(/("token\\":\\")\w+/, '\1[FILTERED]')
+      end
+
+      def store(payment_method, options = {})
+        post = {}
+        MultiResponse.run do |r|
+          if payment_method.is_a?(NetworkTokenizationCreditCard)
+            response = r.process { verify(payment_method, options) }
+            break response unless response.success?
+
+            response.params['source']['customer'] = r.params['customer']
```

Comment:
💬 Comment:

Two things that are confusing unless you have context and know what you are reading:

1) response shadowing a method called also `response`
2) in line 114 you are using `response` and `r` to just level down `the customer` it has two effects:
  a) the original response is going the have `customer` at the top level and also in the `source` key.
  b) is very misleading using `response` and `r`, at this point, both point to the same object => the primary response of the Multiresponse.

Reply:
2. The idea is to have the same information displayed in a normal store response, that's why `r.params[:source]` is that I'll return in the response. Because in a store response, you'll see the same information in the `verify.params[:source]` except by the customer that is on a different level.